src/modules/ekf2/ekf2_main.cpp

```diff
@@ -1422,37 +1422,27 @@ void Ekf2::run()
 					lpos.hagl_max = INFINITY;
 				}
 
-				// Get covariances to vehicle odometry
-				float covariances[24];
-				_ekf.get_covariances(covariances);
+				// get pose covariance and copy the URT to the uORB message
+				float *pos_p = &odom.pose_covariance[0];
 
-				// get the covariance matrix size
-				const size_t POS_URT_SIZE = sizeof(odom.pose_covariance) / sizeof(odom.pose_covariance[0]);
-				const size_t VEL_URT_SIZE = sizeof(odom.velocity_covariance) / sizeof(odom.velocity_covariance[0]);
-
-				// initially set pose covariances to 0
-				for (size_t i = 0; i < POS_URT_SIZE; i++) {
-					odom.pose_covariance[i] = 0.0;
+				for (unsigned x = 0; x < 6; x++) {
+					for (unsigned y = x; y < 6; y++) {
+						*pos_p++ = _ekf.pose_covariances()(x, y);
```

Comment:
Can you move the function call `_ekf.pose_covariances()` out of the loop into a separate variable?
The compiler might do that, but it does not have to.

Reply:
@bkueng done in 9c92002.